### ISSUE TITLE
Cannot read property 'findDeclaration' of undefined

### ISSUE THREAD

**Issue by piotr-cz** [rollup/rollup#1198]
Since update to _v0.38.2_ I've started to receive following fatal error:
`Cannot read property 'findDeclaration' of undefined`.

Only thing I've been able to confrim is that the scope which is passed callHasEffects function is indeed undefined.

Issue persists in _v0.38.3_.

<details>
  <summary>Stack trace</summary>
TypeError: Cannot read property 'findDeclaration' of undefined
    at callHasEffects (xxx\node_modules\rollup\src\ast\nodes\shared\callHasEffects.js:75:29)
    at CallExpression.hasEffects (xxx\node_modules\rollup\src\ast\nodes\CallExpression.js:30:10)
    at CallExpression.isUsedByBundle (xxx\node_modules\rollup\src\ast\nodes\CallExpression.js:41:15)
    at xxx\node_modules\rollup\src\Bundle.js:161:31
</details>

**Comment by kzc** [comment]
Could you post a small javascript example that triggers this error?

**Comment by sunjay** [comment]
I'm also experiencing this problem. There was no error until I updated to the latest rollup. The stack trace also doesn't point to any of my code so there is nothing I can provide to reproduce the problem without some more information from the rollup team about what to look for.

```
Cannot read property 'findDeclaration' of undefined
TypeError: Cannot read property 'findDeclaration' of undefined
    at callHasEffects (./node_modules/rollup/src/ast/nodes/shared/callHasEffects.js:75:29)
    at CallExpression.hasEffects (./node_modules/rollup/src/ast/nodes/CallExpression.js:34:10)
    at CallExpression.isUsedByBundle (./node_modules/rollup/src/ast/nodes/CallExpression.js:45:15)
    at ./node_modules/rollup/src/Bundle.js:162:31
```

**Comment by sunjay** [comment]
Rolling back to 0.38.1 fixes the problem for the time being. That means something something happened between 0.38.1 and 0.38.2 to cause this.

**Comment by yamafaktory** [comment]
Hi, I am also experiencing the same issue after a Greenkeeper PR. You can check it here: https://github.com/yamafaktory/buble-react-rollup-starter/pull/8.

**Comment by kzc** [comment]
Repro thanks to @yamafaktory 

Adapted from `react-dom/lib/validateDOMNesting.js` in production mode:

rollup version 0.39.0

$ cat repro.js 
```js
if (0) {
    var inScopeTags = [];
    var buttonScopeTags = inScopeTags.concat(['button']);
}
```
$ node bin/rollup repro.js 
```
Cannot read property 'findDeclaration' of undefined
```
Regression introduced in commit https://github.com/rollup/rollup/commit/b28737069e5df3e05ed9e1174b2bc4d5897d0e66

**Comment by piotr-cz** [comment]
I'm using `react` and `react-dom` as well.
It might indeed be https://github.com/rollup/rollup/commit/b28737069e5df3e05ed9e1174b2bc4d5897d0e66 because it was merged into v0.38.2

**Comment by Rich-Harris** [comment]
Thanks all – fixed in 0.39.1
